_posts/2018-09-12-graphql-is-not-odata.md

```diff
@@ -0,0 +1,224 @@
+
+<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">GraphQL is not OData. Can we please stop casting OData’s flaws onto GraphQL?<br><br>At Microsoft, I used and propagated OData against my own judgment. I’m sorry I couldn’t stop the train wreck.<br><br>At Concur, we successfully employ GraphQL over top of RESTful APIs.<br><br>Feel free to @ me.</p>&mdash; Jeff Handley (@JeffHandley) <a href="https://twitter.com/JeffHandley/status/1034643959434293248?ref_src=twsrc%5Etfw">August 29, 2018</a></blockquote>
+
+## Experience with OData
+
+I spent several years at Microsoft. I contributed to a few projects you might be familiar with:
+
+* Silverlight UI SDK
+* WCF RIA Services
+* .NET Framework
+* Visual Studio
+* ASP.NET Web Pages
+* Razor
+* NuGet and the NuGet Gallery
+
+### WCF RIA Services
+
+I had my first experience with OData while working on WCF RIA Services. I started as an SDE II on that project but eventually became the dev manager. WCF RIA Services brought Silverlight and ASP.NET together, allowing you to build Line-of-Business applications with ASP.NET-based domain/business layers and Silverlight-based UIs. This combination made it pretty easy to grow from forms-over-data to business-logic heavy applications, keeping the business logic out of the UI through a clear separation of concerns.
+
+WCF RIA Services allowed you to either expose your **domain model** directly to the client or instead create a **view model** that would be exposed. By authoring CRUDE (create, read, update, delete, execute) operations and annotating them, you could easily create an API for your client to consume. But we needed a protocol for serializing requests and responses for those CRUDE operations and the data to go along with them. We ultimately landed on using some WCF primitives.
+
+Before we solidifed our decision that we'd use WCF, our project was just called "RIA Services", where RIA stood for "Rich Internet Application." Using WCF resulted in prepending our project name with WCF and, believe it or not, getting reorged into the WCF group. This was all good though--using WCF was far better than using our proprietary protocol developed in our early preview releases.
+
+WCF RIA Services was pretty well engineered. We were promoting a separation of concerns for applications built on our framework, and we ourselves kept our concerns separated. This led to both the server and the client being decoupled from the communication protocol. We were able to swap out that layer independently--largely because Wilco Bauer is amazing. Wilco swapped out our proprietary protocol for WCF in days. And we demonstrated that as we could support multiple protocols in parallel.
+
+Along came OData. Microsoft was pushing OData pretty hard for a while. Some really sharp folks invented it and it had a lot of muscle behind it. It became clear that we needed to support OData for WCF RIA Services. After some long debates, we ultimately landed on the decision of:
```

Comment:
IIRC, OData came from Pablo Castro and his associated group in SQL Server. That says a lot about it's design bent.